In [1]:
import pandas as pd

# Modèle 1 : freezing

In [2]:
df1 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset004_LA_CT00_corrected_voted/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_1er_test/fold_0/validation/patients_history_metrics.csv', index_col='Unnamed: 0')
df1 = df1[~df1['Dice_auricle'].isna()]

df1['New_dice_auricle'].mean()

0.9123193048289298

In [3]:
df_opa1 = df1[df1['LAA_opacification_defect']==1]
df_opa1['New_dice_auricle'].mean()

0.876944817837498

In [4]:
df_no_opa1 = df1[df1['LAA_opacification_defect']==0]
df_no_opa1['New_dice_auricle'].mean()

0.936639264635539

# Modèle 2 : pas de freezing, 1000 époques

In [5]:
df2 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset004_LA_CT00_corrected_voted/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres/fold_0/prediction_1000/patients_history_metrics.csv', index_col='Unnamed: 0')
df2 = df2[~df2['Dice_auricle'].isna()]

df2['New_dice_auricle'].mean()

0.9495271146303258

In [6]:
df_opa2 = df2[df2['LAA_opacification_defect']==1]
df_opa2['New_dice_auricle'].mean()

0.9332100984169088

In [7]:
df_no_opa2 = df2[df2['LAA_opacification_defect']==0]
df_no_opa2['New_dice_auricle'].mean()

0.9607450632770497

# Modèle 3 : lr_06_1500_epochs

In [8]:
df3 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset004_LA_CT00_corrected_voted/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_lr_06_1500_epochs/fold_0/validation/patients_history_metrics.csv', index_col='Unnamed: 0')
df3 = df3[~df3['Dice_auricle'].isna()]

df3['New_dice_auricle'].mean()

0.9498713772847118

In [9]:
df_opa3 = df3[df3['LAA_opacification_defect']==1]
df_opa3['New_dice_auricle'].mean()

0.9351265603037262

In [10]:
df_no_opa3 = df3[df3['LAA_opacification_defect']==0]
df_no_opa3['New_dice_auricle'].mean()

0.9600084389591395

# Modèle 4 : suite entrainement modèle 2 jusqu’à 1500 époques

In [11]:
df4 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset004_LA_CT00_corrected_voted/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres/fold_0/validation/patients_history_metrics.csv', index_col='Unnamed: 0')
df4 = df4[~df4['Dice_auricle'].isna()]

df4['New_dice_auricle'].mean()

0.9495276191032155

In [12]:
df_opa4 = df4[df4['LAA_opacification_defect']==1]
df_opa4['New_dice_auricle'].mean()

0.933210356733225

In [13]:
df_no_opa4 = df4[df4['LAA_opacification_defect']==0]
df_no_opa4['New_dice_auricle'].mean()

0.9607457369825841

# Modèle 5 : region training on labels, prédiction 500 (avant overfitting)

In [17]:
df5 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset005_LA_CT00_corrected_voted_region/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_model_5/fold_0/prediction_500/patients_history_metrics.csv', index_col='Unnamed: 0')
df5 = df5[~df5['Dice_auricle'].isna()]

df5['New_dice_auricle'].mean()

0.948144704335795

In [18]:
df_opa5 = df5[df5['LAA_opacification_defect']==1]
df_opa5['New_dice_auricle'].mean()

0.936854977922195

In [19]:
df_no_opa5 = df5[df5['LAA_opacification_defect']==0]
df_no_opa5['New_dice_auricle'].mean()

0.9559063912451451

In [20]:
# RELANCER CALCUL DE METRIQUES AVEC NOUVELLE LABELLISATION
# PUIS CALCULER METRIQUES SUR LES LABELS

In [21]:
df_results = pd.DataFrame(index=['all', 'only_defect', 'without_defect'], data={'Modèle 1': [df1['New_dice_auricle'].mean(), df_opa1['New_dice_auricle'].mean(), df_no_opa1['New_dice_auricle'].mean()],
                                                  'Modèle 2': [df2['New_dice_auricle'].mean(), df_opa2['New_dice_auricle'].mean(), df_no_opa2['New_dice_auricle'].mean()],
                                                  'Modèle 3': [df3['New_dice_auricle'].mean(), df_opa3['New_dice_auricle'].mean(), df_no_opa3['New_dice_auricle'].mean()],
                                                  'Modèle 4': [df4['New_dice_auricle'].mean(), df_opa4['New_dice_auricle'].mean(), df_no_opa4['New_dice_auricle'].mean()],
                                                  'Modèle 5': [df5['New_dice_auricle'].mean(), df_opa5['New_dice_auricle'].mean(), df_no_opa5['New_dice_auricle'].mean()],
                                                 })

In [22]:
df_results.style.background_gradient(axis=1, cmap="Blues", subset=['Modèle 2', 'Modèle 3', 'Modèle 4', 'Modèle 5'])


# Résultats sur les labels


Labels :
Overfitting vers époque 500
époque 500 : [0.97, 0.9639, 0.8121, 0.5043]
Best avant 500 (avant overfitting) : époque 490 : [0.9719, 0.9644, 0.8582, 0.6213]
Best = époque 1278 : [0.9768, 0.9675, 0.8434, 0.611]

In [23]:
#pseudo dice (from logs)
df_labels = pd.DataFrame(index=['epoch 500', 'best epoch before 500 = 490', 'best epoch = 1278'],
                         data={'whole_atrium': [0.97, 0.9719, 0.9768],
                         'body_and_auricle': [0.9639, 0.9644, 0.9675],
                         'auricle': [0.8121, 0.8582, 0.8434],
                         'opacification_defect': [0.5043, 0.6213, 0.611]})

In [24]:
df_labels.style.background_gradient(axis=0, cmap="Blues")


# Sélection d'un exemple

In [25]:
df_opa3.sort_values('New_dice_auricle')

ID  Stroke_ait_history  peripheral_embolism_hist  embolism_hist  \
Unnamed: 0                                                                     
64          106                   1                         0              1   
38           65                   0                         0              0   
31           49                   1                         0              1   
128         277                   0                         0              0   
19           31                   0                         0              0   
81          155                   1                         0              1   
73          116                   0                         0              0   
75          120                   1                         0              1   
129         278                   1                         0              1   
95          239                   1                         0              1   
100         244                   0                         0              0   

            verified_thrombus  LAA_opacification_defect      Dice       IoU  \
Unnamed: 0                                                                    
64                          0                         1  0.974749  0.950741   
38                          0                         1  0.947503  0.900243   
31                          0                         1  0.944824  0.895418   
128                         0                         1  0.986529  0.973416   
19                          0                         1  0.971907  0.945350   
81                          0                         1  0.976713  0.954487   
73                          0                         1  0.979270  0.959381   
75                          0                         1  0.982933  0.966439   
129                         1                         1  0.981799  0.964249   
95                          0                         1  0.982082  0.964795   
100                         0                         1  0.984752  0.969962   

            nnUnet_id  Contour_Volume  New_dice_auricle  Dice_auricle  \
Unnamed: 0                                                              
64               65.0        120863.0          0.866940      0.981621   
38               39.0        272530.0          0.874538      0.986462   
31               32.0        253115.0          0.901659      0.985566   
128             128.0         97844.0          0.943789      0.981167   
19               19.0         80269.0          0.947919      0.993225   
81               82.0        122657.0          0.950907      0.980693   
73               74.0         92372.0          0.956425      0.989339   
75               76.0        196329.0          0.959772      0.993067   
129             129.0        229647.0          0.959976      0.989690   
95               95.0        191697.0          0.961118      0.994606   
100             100.0        114812.0          0.963348      0.984864   

            Dice_corps  Dice_RSPV_1  Dice_RIPV_2  Dice_LIPV_3  Dice_LSPV_4  \
Unnamed: 0                                                                   
64            0.997683     0.998608     0.987919     0.985140     0.994002   
38            0.986890     0.983386     0.913315     0.855319     0.884921   
31            0.988625     0.865823     0.839642     0.973799     0.951853   
128           0.997044     0.987435     0.958943     0.970299     0.985802   
19            0.993870     0.995070     0.984055     0.994441     0.982800   
81            0.992027     0.975721     0.967090     0.979162     0.882457   
73            0.997895     0.965717     0.983575     0.966854     0.990788   
75            0.998590     0.999191     0.997435     0.999159     0.994227   
129           0.997964     0.994708     0.985142     0.970947     0.989640   
95            0.997352     0.997559     0.983908     0.991713     0.995463   
100           0.997653     0.985397     0.996882     0.997

Si l'on prend la médiane : New_dice_auricle = 0.950907, cela correspond au cas 155_82.

In [26]:
df_opa5.sort_values('New_dice_auricle')

ID  Stroke_ait_history  peripheral_embolism_hist  embolism_hist  \
Unnamed: 0                                                                     
64          106                   1                         0              1   
38           65                   0                         0              0   
31           49                   1                         0              1   
19           31                   0                         0              0   
128         277                   0                         0              0   
81          155                   1                         0              1   
73          116                   0                         0              0   
129         278                   1                         0              1   
75          120                   1                         0              1   
100         244                   0                         0              0   
95          239                   1                         0              1   

            verified_thrombus  LAA_opacification_defect      Dice       IoU  \
Unnamed: 0                                                                    
64                          0                         1  0.973968  0.949256   
38                          0                         1  0.934953  0.877852   
31                          0                         1  0.938677  0.884441   
19                          0                         1  0.974211  0.949719   
128                         0                         1  0.985606  0.971620   
81                          0                         1  0.970101  0.941938   
73                          0                         1  0.967659  0.937344   
129                         1                         1  0.978762  0.958408   
75                          0                         1  0.982729  0.966045   
100                         0                         1  0.978903  0.958678   
95                          0                         1  0.980030  0.960843   

            nnUnet_id  Contour_Volume  New_dice_auricle  Dice_auricle  \
Unnamed: 0                                                              
64               65.0        124336.0          0.873901      0.982379   
38               39.0        340689.0          0.875559      0.984569   
31               32.0        280090.0          0.904196      0.978531   
19               19.0         72818.0          0.945894      0.992193   
128             128.0        104567.0          0.949724      0.979390   
81               82.0        157873.0          0.950003      0.973944   
73               74.0        142174.0          0.955261      0.989801   
129             129.0        266849.0          0.957366      0.988203   
75               76.0        198416.0          0.960178      0.989149   
100             100.0        159469.0          0.964163      0.985845   
95               95.0        213677.0          0.969159      0.993834   

            Dice_corps  Dice_RSPV_1  Dice_RIPV_2  Dice_LIPV_3  Dice_LSPV_4  \
Unnamed: 0                                                                   
64            0.996746     0.981272     0.973180     0.968850     0.982804   
38            0.985907     0.959530     0.911827     0.848394     0.881094   
31            0.985168     0.856968     0.807282     0.943704     0.936856   
19            0.992539     0.988157     0.967581     0.947042     0.945215   
128           0.996271     0.980488     0.967640     0.977086     0.976321   
81            0.991364     0.988233     0.902955     0.961539     0.843674   
73            0.995519     0.750674     0.936185     0.954591     0.980650   
129           0.996983     0.986906     0.937748     0.936363     0.965393   
75            0.997746     0.995851     0.995133     0.943715     0.985298   
100           0.997354     0.967632     0.994152     0.992853     0.994516   
95            0.996475     0.992376     0.945520     0.978

In [29]:
df5[['ID', 'LAA_opacification_defect']]

ID  LAA_opacification_defect
Unnamed: 0                               
7            16                         0
14           25                         0
19           31                         1
24           40                         0
31           49                         1
36           61                         0
38           65                         1
43           74                         0
64          106                         1
73          116                         1
75          120                         1
78          128                         0
81          155                         1
86          230                         0
93          237                         0
95          239                         1
100         244                         1
101         245                         0
103         247                         0
105         251                         0
116         262                         0
117         263                         0
118         264                         0
120         267                         0
126         275                         0
128         277                         1
129         278                         1

In [30]:
df5

ID  Stroke_ait_history  peripheral_embolism_hist  embolism_hist  \
Unnamed: 0                                                                     
7            16                   0                         0              0   
14           25                   1                         0              1   
19           31                   0                         0              0   
24           40                   0                         0              0   
31           49                   1                         0              1   
36           61                   0                         0              0   
38           65                   0                         0              0   
43           74                   0                         0              0   
64          106                   1                         0              1   
73          116                   0                         0              0   
75          120                   1                         0              1   
78          128                   0                         0              0   
81          155                   1                         0              1   
86          230                   1                         0              1   
93          237                   1                         0              1   
95          239                   1                         0              1   
100         244                   0                         0              0   
101         245                   1                         0              1   
103         247                   0                         0              0   
105         251                   0                         0              0   
116         262                   0                         0              0   
117         263                   0                         0              0   
118         264                   0                         0              0   
120         267                   1                         0              1   
126         275                   0                         0              0   
128         277                   0                         0              0   
129         278                   1                         0              1   

            verified_thrombus  LAA_opacification_defect      Dice       IoU  \
Unnamed: 0                                                                    
7                           0                         0  0.978835  0.958547   
14                          0                         0  0.954547  0.913046   
19                          0                         1  0.974211  0.949719   
24                          0                         0  0.962494  0.927700   
31                          0                         1  0.938677  0.884441   
36                          0                         0  0.954134  0.912290   
38                          0                         1  0.934953  0.877852   
43                          0                         0  0.975956  0.953040   
64                          0                         1  0.973968  0.949256   
73                          0                         1  0.967659  0.937344   
75                          0                         1  0.982729  0.966045   
78                          0                         0  0.962929  0.928509   
81                          0                         1  0.970101  0.941938   
86                          0                         0  0.981826  0.964300   
93                          0                         0  0.981431  0.963538   
95                          0                         1  0.980030  0.960843   
100                         0                         1  0.978903  0.958678   
101                         0                         0  0.977944  0.956840   
103                         0                         0  0.986808  0.973959   
105                         0    

In [2]:
import pandas as pd

In [2]:
df6 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset005_LA_CT00_corrected_voted_region/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_500_epochs/fold_0/validation_best_500/patients_history_metrics_etude_corr.csv', index_col='Unnamed: 0')
df6 = df6[~df6['Dice_auricle'].isna()]

df6['New_dice_auricle'].mean()

0.939155818010795

In [3]:
df_opa6 = df6[df6['LAA_opacification_defect']==1]
df_opa6['New_dice_auricle'].mean()

0.9215811219646425

In [4]:
df_no_opa6 = df6[df6['LAA_opacification_defect']==0]
df_no_opa6['New_dice_auricle'].mean()

0.9512384215425249

In [15]:
df6 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset005_LA_CT00_corrected_voted_region/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_500_epochs/fold_0/validation_best_500/patients_history_metrics.csv', index_col='Unnamed: 0')
df6 = df6[~df6['Dice_auricle'].isna()]

df6['New_dice_auricle'].mean()

0.9465609439832475

In [16]:
df_opa6 = df6[df6['LAA_opacification_defect']==1]
df_opa6['New_dice_auricle'].mean()

0.9383105515169963

In [17]:
df_no_opa6 = df6[df6['LAA_opacification_defect']==0]
df_no_opa6['New_dice_auricle'].mean()

0.9522330888037949

In [5]:
'/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres/fold_0/training_1000_epochs/validation/patients_history_metrics.csv'

'/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres/fold_0/training_1000_epochs/validation/patients_history_metrics.csv'

In [6]:
df7 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres/fold_0/training_1000_epochs/validation/patients_history_metrics.csv', index_col='Unnamed: 0')
df7 = df7[~df7['Dice_auricle'].isna()]

df7['New_dice_auricle'].mean()

0.9504882180003089

In [7]:
df_opa7 = df7[df7['LAA_opacification_defect']==1]
df_opa7['New_dice_auricle'].mean()

0.9420598054257269

In [8]:
df_no_opa7 = df7[df7['LAA_opacification_defect']==0]
df_no_opa7['New_dice_auricle'].mean()

0.956282751645334

In [6]:
df8 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_classic_loss/fold_0/training_1000_epochs/validation_872_best_1000/patients_history_metrics_etude_corr.csv', index_col='Unnamed: 0')
df8 = df8[~df8['Dice_auricle'].isna()]

df8['New_dice_auricle'].mean()

0.9441144532400729

In [12]:
df_opa8 = df8[df8['LAA_opacification_defect']==1]
df_opa8['New_dice_auricle'].mean()

0.9434546054730123

In [13]:
df_no_opa8 = df8[df8['LAA_opacification_defect']==0]
df_no_opa8['New_dice_auricle'].mean()

0.952262404725617

In [4]:
df9 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_classic_loss/fold_0/validation/patients_history_metrics.csv', index_col='Unnamed: 0')
df9 = df9[~df9['Dice_auricle'].isna()]

df9['New_dice_auricle'].mean()

0.9518168326045249

In [5]:
df_opa9 = df9[df9['LAA_opacification_defect']==1]
df_opa9['New_dice_auricle'].mean()

0.942147331389574

In [6]:
df_no_opa9 = df9[df9['LAA_opacification_defect']==0]
df_no_opa9['New_dice_auricle'].mean()

0.9584646146898035

In [7]:
df10 = pd.read_csv('/media/sharedata/atriumCT/atrium_nnunet/nnUNet_results/Dataset006_LA_CT00_region_fenetrage/nnUNetTrainer__nnUNetPlans_Dataset002__3d_fullres_classic_loss/fold_0/validation/patients_history_metrics_etude_corr.csv', index_col='Unnamed: 0')
df10 = df10[~df10['Dice_auricle'].isna()]

df10['New_dice_auricle'].mean()

0.9473920179612799

In [24]:
df_opa10 = df10[df10['LAA_opacification_defect']==1]
df_opa10['New_dice_auricle'].mean()

0.9317712387389482

In [25]:
df_no_opa10 = df10[df10['LAA_opacification_defect']==0]
df_no_opa10['New_dice_auricle'].mean()

0.958131303676633

In [8]:
df9['Contour_Volume'].mean()

247593.1851851852

In [14]:
df8['Contour_Volume'].mean()

205460.2962962963

In [8]:
df9['Contour_Volume'].mean()

247593.1851851852

In [18]:
df6['Contour_Volume'].mean()

275510.25925925927

In [19]:
df_opa6[['ID', 'New_dice_auricle']]

ID  New_dice_auricle
Unnamed: 0                       
19           31          0.945258
31           49          0.903359
38           65          0.875960
64          106          0.880748
73          116          0.960231
75          120          0.959429
81          155          0.957388
95          239          0.968327
100         244          0.955312
128         277          0.957480
129         278          0.957922

In [20]:
df_opa7[['ID', 'New_dice_auricle']]

ID  New_dice_auricle
Unnamed: 0                       
19           31          0.960525
31           49          0.912970
38           65          0.867533
64          106          0.867212
73          116          0.969298
75          120          0.967809
81          155          0.958499
95          239          0.974208
100         244          0.956448
128         277          0.962443
129         278          0.965713

In [24]:
df_opa5[['ID', 'New_dice_auricle']]

ID  New_dice_auricle
Unnamed: 0                       
19           31          0.945894
31           49          0.904196
38           65          0.875559
64          106          0.873901
73          116          0.955261
75          120          0.960178
81          155          0.950003
95          239          0.969159
100         244          0.964163
128         277          0.949724
129         278          0.957366

In [12]:
df_opa8[['ID', 'New_dice_auricle']]

ID  New_dice_auricle
Unnamed: 0                       
19           31          0.956223
31           49          0.914265
38           65          0.878029
64          106          0.879876
73          116          0.968123
75          120          0.965029
81          155          0.960275
95          239          0.968063
100         244          0.950890
128         277          0.965981
129         278          0.971247

In [20]:
df_opa5['New_dice_auricle'].mean()

0.936854977922195

In [16]:
df_opa8['New_dice_auricle'].mean()

0.9434546054730123

In [1]:
import pandas as pd

In [2]:
# Transformer
df11 = pd.read_csv('/media/sharedata/atriumCT/atrium_medFormer/results/LA_CT00/fold_0_val/mha_files/patients_history_metrics_etude_corr.csv')
df11 = df11[~df11['Dice_auricle'].isna()]

df11['New_dice_auricle'].mean()

0.8685421090326549

In [4]:
df11

ID  medformer_id      Dice       IoU  Contour_Volume  New_dice_auricle  \
0   105            64  0.214196  0.119944       2612071.0          0.404490   
1   118            75  0.864913  0.761979        418883.0          0.935195   
2    14             6  0.908695  0.832668        271721.0          0.861749   
3   150            81  0.946809  0.898991        203393.0          0.826021   
4    16             7  0.955319  0.914460        206923.0          0.941780   
5    21            10  0.937328  0.882048        127495.0          0.891686   
6    23            12  0.873712  0.775744        397238.0          0.797136   
7   240            96  0.968280  0.938511        281775.0          0.958672   
8   242            98  0.954271  0.912542        432545.0          0.876977   
9   243            99  0.930883  0.870703        513632.0          0.790060   
10  245           101  0.966349  0.934889        362961.0          0.944245   
11  246           102  0.949122  0.903170        449093.0          0.953918   
12   24            13  0.957471  0.918411         88111.0          0.935431   
13  255           109  0.767360  0.622534       1396731.0          0.828587   
14  274           125  0.953422  0.910989        258720.0          0.898435   
15  275           126  0.962113  0.926991        532124.0          0.918150   
16   29            17  0.932907  0.874251        340180.0          0.913893   
17   31            19  0.908862  0.832948        251225.0          0.901943   
18   43            27  0.891310  0.803930        329554.0          0.923933   

    Dice_auricle  Dice_corps  Dice_RSPV_1  Dice_RIPV_2  Dice_LIPV_3  \
0       0.519184    0.176244     0.027517     0.370683     0.619984   
1       0.969131    0.906684     0.662308     0.629936     0.317503   
2       0.898244    0.971278     0.892967     0.453006     0.636173   
3       0.796846    0.981736     0.944403     0.720891     0.836126   
4       0.952358    0.989391     0.858734     0.882400     0.948839   
5       0.926360    0.971437     0.684226     0.769537     0.828973   
6       0.776020    0.953512     0.787963     0.743133     0.760783   
7       0.973991    0.990416     0.898645     0.930159     0.854253   
8       0.802904    0.970061     0.977629     0.821435     0.958662   
9       0.570206    0.975137     0.924546     0.616527     0.824294   
10      0.964083    0.988736     0.951778     0.827121     0.908303   
11      0.988275    0.981559     0.919938     0.789081     0.627884   
12      0.944318    0.978182     0.903446     0.923895     0.891955   
13      0.852843    0.829647     0.307340     0.136028     0.440205   
14      0.819412    0.973165     0.842136     0.761308     0.789130   
15      0.940480    0.986658     0.961156     0.655913     0.703809   
16      0.858800    0.977846     0.862741     0.901621     0.941655   
17      0.895527    0.975541     0.627576     0.743150     0.509083   
18      0.904946    0.976055     0.879204     0.943377     0.907903   

    Dice_LSPV_4  Contour_vol_auricle  Spacing_ref_0  Spacing_ref_1  \
0      0.013709             256516.0       0.322266       0.322266   
1      0.913753              28871.0       0.427734       0.427734   
2      0.853788              69743.0       0.531250       0.531250   
3      0.950817              80676.0       0.371094       0.371094   
4      0.946264              32180.0       0.451172       0.451172   
5      0.931113              19036.0       0.486328       0.486328   
6      0.747687             112397.0       0.513672       0.513672   
7      0.869806              62019.0       0.306641       0.306641   
8      0.956076             111509.0       0.429688       0.429688   
9      0.956979             227227.0       0.408203       0.408203   
10     0.948843              88626.0       0.371094       0.371094   
11     0.904373              51696.0       0.322266       0.322266   
12     0.916732              11703.0       0.435547       0.435547   
13     0.255280        

In [6]:
df11['Dice'].mean()

0.8864905482827794

In [3]:
df10

NameError: name 'df10' is not defined

In [4]:
df11

ID  medformer_id      Dice       IoU  Contour_Volume  New_dice_auricle  \
0   105            64  0.214196  0.119944       2612071.0          0.404490   
1   118            75  0.864913  0.761979        418883.0          0.935195   
2    14             6  0.908695  0.832668        271721.0          0.861749   
3   150            81  0.946809  0.898991        203393.0          0.826021   
4    16             7  0.955319  0.914460        206923.0          0.941780   
5    21            10  0.937328  0.882048        127495.0          0.891686   
6    23            12  0.873712  0.775744        397238.0          0.797136   
7   240            96  0.968280  0.938511        281775.0          0.958672   
8   242            98  0.954271  0.912542        432545.0          0.876977   
9   243            99  0.930883  0.870703        513632.0          0.790060   
10  245           101  0.966349  0.934889        362961.0          0.944245   
11  246           102  0.949122  0.903170        449093.0          0.953918   
12   24            13  0.957471  0.918411         88111.0          0.935431   
13  255           109  0.767360  0.622534       1396731.0          0.828587   
14  274           125  0.953422  0.910989        258720.0          0.898435   
15  275           126  0.962113  0.926991        532124.0          0.918150   
16   29            17  0.932907  0.874251        340180.0          0.913893   
17   31            19  0.908862  0.832948        251225.0          0.901943   
18   43            27  0.891310  0.803930        329554.0          0.923933   

    Dice_auricle  Dice_corps  Dice_RSPV_1  Dice_RIPV_2  Dice_LIPV_3  \
0       0.519184    0.176244     0.027517     0.370683     0.619984   
1       0.969131    0.906684     0.662308     0.629936     0.317503   
2       0.898244    0.971278     0.892967     0.453006     0.636173   
3       0.796846    0.981736     0.944403     0.720891     0.836126   
4       0.952358    0.989391     0.858734     0.882400     0.948839   
5       0.926360    0.971437     0.684226     0.769537     0.828973   
6       0.776020    0.953512     0.787963     0.743133     0.760783   
7       0.973991    0.990416     0.898645     0.930159     0.854253   
8       0.802904    0.970061     0.977629     0.821435     0.958662   
9       0.570206    0.975137     0.924546     0.616527     0.824294   
10      0.964083    0.988736     0.951778     0.827121     0.908303   
11      0.988275    0.981559     0.919938     0.789081     0.627884   
12      0.944318    0.978182     0.903446     0.923895     0.891955   
13      0.852843    0.829647     0.307340     0.136028     0.440205   
14      0.819412    0.973165     0.842136     0.761308     0.789130   
15      0.940480    0.986658     0.961156     0.655913     0.703809   
16      0.858800    0.977846     0.862741     0.901621     0.941655   
17      0.895527    0.975541     0.627576     0.743150     0.509083   
18      0.904946    0.976055     0.879204     0.943377     0.907903   

    Dice_LSPV_4  Contour_vol_auricle  Spacing_ref_0  Spacing_ref_1  \
0      0.013709             256516.0       0.322266       0.322266   
1      0.913753              28871.0       0.427734       0.427734   
2      0.853788              69743.0       0.531250       0.531250   
3      0.950817              80676.0       0.371094       0.371094   
4      0.946264              32180.0       0.451172       0.451172   
5      0.931113              19036.0       0.486328       0.486328   
6      0.747687             112397.0       0.513672       0.513672   
7      0.869806              62019.0       0.306641       0.306641   
8      0.956076             111509.0       0.429688       0.429688   
9      0.956979             227227.0       0.408203       0.408203   
10     0.948843              88626.0       0.371094       0.371094   
11     0.904373              51696.0       0.322266       0.322266   
12     0.916732              11703.0       0.435547       0.435547   
13     0.255280        

In [9]:
df = pd.read_csv(f'/media/sharedata/atriumCT/atrium_nnunet/raw_data/Dataset004_LA_CT00_corrected_voted/prediction_3d_dataset002/patients_history.csv')

In [10]:
df

Unnamed: 0   ID  Stroke_ait_history  peripheral_embolism_hist  \
0             0    6                   1                         0   
1             1    7                   1                         0   
2             2    8                   0                         0   
3             3    9                   1                         0   
4             4   11                   0                         0   
..          ...  ...                 ...                       ...   
127         127  276                   0                         0   
128         128  277                   0                         0   
129         129  278                   1                         0   
130         130  282                   0                         0   
131         131  321                   0                         0   

     embolism_hist  verified_thrombus  LAA_opacification_defect      Dice  \
0                1                  0                         0  0.902653   
1                1                  0                         1  0.908900   
2                0                  0                         0  0.963089   
3                1                  0                         1  0.980152   
4                0                  0                         0       NaN   
..             ...                ...                       ...       ...   
127              0                  1                         1  0.917019   
128              0                  0                         1  0.944672   
129              1                  1                         1  0.940356   
130              0                  0                         0  0.950137   
131              0                  0                         0  0.946245   

          IoU  nnUnet_id  
0    0.822578        0.0  
1    0.833013        1.0  
2    0.928806        2.0  
3    0.961077        3.0  
4         NaN        NaN  
..        ...        ...  
127  0.846754      126.0  
128  0.895145      127.0  
129  0.887427      128.0  
130  0.905010      129.0  
131  0.897975      130.0  

[132 rows x 10 columns]

In [30]:
df11 = df11.set_index('ID').join(df[['ID', 'LAA_opacification_defect']].set_index('ID'), on='ID', how='left')

In [26]:
pd.options.display.max_rows = 999

In [31]:
df_opa11 = df11[df11['LAA_opacification_defect']==1]
df_opa11['New_dice_auricle'].mean()

0.8592477741633937

In [33]:
df_no_opa11 = df11[df11['LAA_opacification_defect']==0]
df_no_opa11['New_dice_auricle'].mean()

0.873963804373057